### ISSUE TITLE
email.utils.getaddresses() incorrect value in email address part of tuple

### ISSUE THREAD

**Issue by michal-dagan** [python/cpython#107919]
# Bug report

## Checklist

- [x] I am confident this is a bug in CPython, not a bug in a third-party project
- [ ] I have searched the CPython issue tracker, and am confident this bug has not been reported before

## A clear and concise description of the bug

I have this 'From' field in my EML file:
`From: Test TEST, test<test@gmail.com>`

  ```py
gel_all_values_from_email_by_entry = eml.get_all(entry, [])
> ['Test TEST, test<test@test.com>']
getaddresses(gel_all_values_from_email_by_entry)
> [('', 'Test'), ('test', 'test@gmail.com')]
```
where 'Test' does not represent an email address

# Your environment

- CPython versions tested on:
- Operating system and architecture:

**Comment by tkyanko** [comment]
If I understand RFC 2822 correctly, `Test TEST` is completely invalid.  Does this seem like something that should throw a `ValueError` or not return an invalid email? It seems like either choice has pros/cons.

**Comment by jpl-jengelke** [comment]
I think what one would expect from this is: 
```
[('Test TEST, test', 'test@gmail.com')]
```
This is because the comma actually represents part of the `display name`.

**Comment by jpl-jengelke** [comment]
I don't think the purpose of `getaddresses()` is to validate addresses. It's just to parse them into a list based on delimiter, and it seems to do that well. The exception is thrown when `Address(addr_spec='TEST')` is instantiated.

What I would expect, though, is that the `display_name` (first item in tuple) is set to `test` if `TEST` becomes the `addr_spec`. So that's missing.

**Comment by fsc-eriker** [comment]
No, the syntax is `display name <terminus>`; the string `Test TEST` is not a valid address (but `Test` alone could be a valid email address in the local domain).

**Comment by jpl-jengelke** [comment]
> No, the syntax is `display name <terminus>`; the string `Test TEST` is not a valid address (but `Test` alone could be a valid email address in the local domain).
Why not? If `TEST` is a valid address, then `TEST` alone (username without domain) is valid and by default I think it usually sets the display name as the username in the absence of a display name. Likewise, since `TEST` in `Test TEST` is the address I believe it splits on the space and assigns `Test` as the display name.

**Comment by fsc-eriker** [comment]
> Why not? If TEST is a valid address, then TEST alone (username without domain) is valid and by default I think it usually sets the display name as the username in the absence of a display name. Likewise, since TEST in Test TEST is the address I believe it splits on the space and assigns Test as the display name.

I can see from your previous comment that you believe that, but based on my reading of RFC5322 I do not believe that to be true. I can't prove a negative; if you can support this interpretation by a reference to the spec, that would be interesting.
